src/DDTrace/Integrations/CakePHP/CakePHPIntegration.php

```diff
@@ -94,15 +90,15 @@ function (SpanData $span, array $args) use ($integration) {
             \DDTrace\trace_method('ExceptionRenderer', '__construct', [
                 'instrument_when_limited' => 1,
                 'posthook' => function (SpanData $span, array $args) use ($integration) {
-                    $integration->rootSpan->setError($args[0]);
+                    $integration->setError($integration->rootSpan, $args[0]);
                     return false;
                 },
             ]);
 
             \DDTrace\trace_method('CakeResponse', 'statusCode', [
                 'instrument_when_limited' => 1,
                 'posthook' => function (SpanData $span, $args, $return) use ($integration) {
-                    $integration->rootSpan->setTag(Tag::HTTP_STATUS_CODE, $return);
+                    $integration->rootSpan->meta[Tag::HTTP_STATUS_CODE] = $return;
```

Comment:
There is a side-effect in current setTag(), is this intentional, or do we want to retain the `Internal Server Error` error type here?
```
            if ($key === Tag::HTTP_STATUS_CODE && $value >= 500) {
                $this->hasError = true;
                if (!isset($this->internalSpan->meta[Tag::ERROR_TYPE])) {
                    $this->internalSpan->meta[Tag::ERROR_TYPE] = 'Internal Server Error';
                }
            }
```

Reply:
Need to check this better but the new exception code should take care of it. I had looked into it a long time ago so I need to go back and look. But good catch as a double-check.

Reply:
About this. I think it is correct that we do not do this adjustment. We already set trace (local root span) on error when the status code is set by PHP and I don't think that further changes are required.

Users that use our legacy manual instrumentation api will have it for free.

Even the message would stay the same.